base/ca/src/org/dogtagpki/server/rest/CAInfoService.java

```diff
@@ -180,11 +183,23 @@ private static KRAInfoClient getKRAInfoClient(KRAConnectorInfo connInfo)
         CMSEngine engine = CMS.getCMSEngine();
         EngineConfig cs = engine.getConfig();
 
+        ICertificateAuthority ca = (ICertificateAuthority) engine.getSubsystem(ICertificateAuthority.ID);
+        ICAService caService = (ICAService) ca.getCAService();
+        IConfigStore kraConnectorConfig = cs.getSubStore(KRAConnectorProcessor.PREFIX);
+
         ClientConfig config = new ClientConfig();
         int port = Integer.parseInt(connInfo.getPort());
         config.setServerURL("https", connInfo.getHost(), port);
         config.setNSSDatabase(cs.getInstanceDir() + "/alias");
 
+        // Use client cert specified in KRA connector (i.e. subsystem cert)
+        String nickname = kraConnectorConfig.getString("nickName", null);
+        if (nickname == null) {
+            // Use CA signing cert as client certificate
```

Comment:
same here.  Let it fail instead.

Reply:
Agreed. Is the suggested change above OK?